### PR TITLE
CA-305383: Do not save unchanged Snapshot Schedule information.

### PR DESCRIPTION
**Pull request by AaronRobson**
With the new API bindings reflecting this change to the Helper.AreEqual2 function: https://github.com/xapi-project/xen-api-sdk/pull/156
The PR includes the new Bindings and a second commit which unit tests the change.

### CODE REVIEW

File: XenAdminTests/XenModelTests/AreEqual2Tests.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+namespace XenAdminTests.XenModelTests
+{
+    [TestFixture, Category(TestCategories.Unit)]
+    public class AreEqual2Tests
+    {
+        [Test]
+        public void A_null_is_considered_equal_to_itself()
+        {
+            List<string> given = null;
+            Assert.That(Helper.AreEqual2(given, given), Is.True);
```
Comment (kc284) [inline XenAdminTests/XenModelTests/AreEqual2Tests.cs]: Simplification suggestion: you can use `Assert.IsTrue`, `Assert.IsFalse`.

File: XenAdminTests/XenModelTests/AreEqual2Tests.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        }
+
+        [Test]
+        public void Separate_nulls_are_considered_equal_to_each_other()
+        {
+            Assert.That(Helper.AreEqual2((List<string>)null, (List<string>)null), Is.True);
```
Comment (kc284) [inline XenAdminTests/XenModelTests/AreEqual2Tests.cs]: Let's add one more assertion for null Dictionaries since they are a class used heavily in the bindings.
Also, an important test case missing is comparison of null List with empty List and null Dictionary with empty Dictionary (which is what differentiates the method AreEqual2 from AreEqual).

Comment (AaronRobson) [reply XenAdminTests/XenModelTests/AreEqual2Tests.cs]: Agreed, its better to be thorough.
